### PR TITLE
Marauder parts and construction steps/sprites

### PR DESCRIPTION
**Pull request by JMWTurner**
mecha

Now you can build them. Well you still need admins to give you the parts.

I think I'll make them require a certain level of nanotrasen tech in the future.

### CODE REVIEW

File: code/game/mecha/mecha_parts.dm [hunk 1]
```diff
@@ -161,6 +161,52 @@
 
 
 
+//////////// Marauder
+
+/obj/item/mecha_parts/chassis/marauder
+	name = "Marauder Chassis"
+
+	New()
```
Comment (ComicIronic) [inline code/game/mecha/mecha_parts.dm]: No relative pathing.

File: code/game/mecha/mecha_wreckage.dm [hunk 2]
```diff
@@ -103,6 +103,23 @@
...
+	var/list/parts = list(
+		/obj/item/mecha_parts/part/marauder_torso,
+		/obj/item/mecha_parts/part/marauder_head,
+		/obj/item/mecha_parts/part/marauder_left_arm,
+		/obj/item/mecha_parts/part/marauder_right_arm,
+		/obj/item/mecha_parts/part/marauder_left_leg,
+		/obj/item/mecha_parts/part/marauder_right_leg,
+		)
+	for(var/i=0;i<2;i++)
+		if(!isemptylist(parts) && prob(40))
```
Comment (ComicIronic) [inline code/game/mecha/mecha_wreckage.dm]: I love how pessimistic this code is.
"Let's make sure the list I just made with more than one elements has at least one element after I maybe take one away."

File: html/changelogs/JMWTurner-MarauderParts.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+author: JMWTurner
+changes: []
```
Comment (ComicIronic) [inline html/changelogs/JMWTurner-MarauderParts.yml]: Good thing he make those ARRAY changes.

**Comment by ghost** [comment]
Is it possible to scan and make blueprints of the mech parts?

If so, I've got ideas for away mission/vault loot, to help make mechanics useful

**Comment by JMWTurner** [comment]
@Skullyton Yeah but don't place marauder parts in away missions/vaults please.

**Comment by JMWTurner** [comment]
Watching this pic inspires me to code programmable assembly-lines machines with conveyor belts to automate the exosuit building process, like in Factorio.

**Comment by ghost** [comment]
@JMWTurner Oh, I wasn't gonna place the parts themselves. Probably have them in a blueprint pool for random loot so it'd be sheer luck to get the full marauder

**Comment by jakmak** [comment]
What if to make the parts via robotics fabricator you would need NT tech from cargo and materials research all the way to 10 meaning if you want this mech you really need to put your back into it.

**Comment by ComicIronic** [comment]
I don't think it should be available at all, we have a tendency to make everything obtainable in a way that devalues loot and exclusive items.

Besides, there's already an end-game mech.

File: code/game/mecha/combat/marauder.dm [hunk 4]
```diff
@@ -58,6 +58,13 @@
 	src.smoke_system.attach(src)
 	return
 
+/obj/mecha/combat/marauder/series/New()//Manually-built marauders have no equipments
+	..()
+	var/obj/item/mecha_parts/mecha_equipment/ME
+	for(ME in equipment)
```
Comment (ComicIronic) [inline code/game/mecha/combat/marauder.dm]: Good god how is this legal

Comment (JMWTurner) [reply code/game/mecha/combat/marauder.dm]: I wanted to avoid doing map changes. Guess I'll have to.

**Comment by JMWTurner** [comment]
I agree with @ComicIronic , I don't want those parts to be simply craftable by R&D, but them simply existing in the code gives a bit of room for admin rewards for dedicated roboticists. Also for completion's sake.

We don't have ways to acquire Dark Ripley parts to my knowledge, yet they do exist, (or at least the construction sprites do)

File: code/game/mecha/combat/marauder.dm (hunk 4 above)
Comment (ComicIronic) [reply code/game/mecha/combat/marauder.dm]: No, I meant the way you declared the var.
Declare it in the loop.

**Pull request merged**